man/man8/zfs-load-key.8

```diff
@@ -166,6 +166,27 @@ will become one. Alternatively, the
 .Fl i
 flag may be provided to cause an encryption root to inherit the parent's key
 instead.
+.Pp
+Changing the user's key does not change the underlying master key used to
+encrypt the dataset.  Therefore, if a user key is compromised,
+.Nm zfs Cm change-key
+does not necessarily protect newly-written data from attack.  The master key
+is compromised if an attacker obtains a user key and the corresponding
+wrapped master key. Currently,
+.Nm zfs Cm change-key
+does not overwrite the previous wrapped master key on disk, so it is
+accessible via forensic analysis for an indeterminate length of time.
+.Pp
+In the event of a master key compromise, ideally the drives should be securely
+erased to remove all the old data (which is readable using the compromised
+master key), a new pool created, and the data copied back. This can be
+approximated in place by creating new datasets, copying the data
+(e.g. using
+.Nm zfs Cm send
+|
+.Nm zfs Cm recv Ns
+), and then clearing the free space with
+.Nm zfs Cm initialize Ns .
```

Comment:
This line is a little problematic because `zpool initialize` unfortunately doesn't guarantee that the old data isnt remapped somewhere else. We really should look into adding secure erase functionality to get us closer to solving this problem.

Reply:
You mean remapped by the underlying storage hardware?  (e.g. an SSD's FTL)

Reply:
yes. or if the underlying device is a file.

Reply:
How about rephrasing this to recommend using `zpool trim --secure` if supported by your hardware.  This will issue "secure discard" commands to the device is the same way `blkdiscard --secure` would.  The `--secure` option will also override the default minimum trim size to ensure every free sector gets trimmed.  Otherwise, `zpool initialize` is recommended for HDDs.

Reply:
Could there be a mechanism added that zeros out the freed keystore after a txg commit? As long as there is no checkpoint.

Reply:
@scineram Yes, that was discussed. Hopefully someone will do that. This PR is about documenting the current limitations.

> How about rephrasing this to recommend using zpool trim --secure if supported by your hardware.

Done.